2_standard_libraries/functools_examples.py

```diff
@@ -0,0 +1,43 @@
+from functools import partial, reduce, lru_cache
+
+
+def do_partial():
+    # Shortens the code, returns partial object which behaves like func with
+    # positional/keyword arguments
+    def divide(a, b):
+        return a / b
+
+    divideby2 = partial(divide, b=2)
+    divideby3 = partial(divide, b=3)
+    result = divideby2(4)
+    result2 = divideby3(1)
+    return result, result2
+
+
+def do_reduce(iterable_data):
+    # Reduce the iterable to single value from LEFT to RIGHT - why is it
+    # useful?
```

Comment:
it's useful in cases when you need an accumulated value of previous elements in your iterable

```
>>> l1 = []
>>> def foo(el):
...     l1.append(el)
...     return el
...
>>> reduce(lambda x, y: foo(x + y), [1, 2, 3, 4, 5])
15
>>> l1
[3, 6, 10, 15]
>>>
```